apps/nsqd/nsqd.go

```diff
@@ -232,6 +238,10 @@ func (p *program) Start() error {
 	}
 	nsqd.Main()
 
+	// hook into addons
+	addons := contrib.NewEnabledNSQDAddons(opts.ModOpt, nsqd, nsqd.Logf)
```

Comment:
Eh, well, not much point in passing the public method `Logf` separately.

Imagine if something like `addons.Start()` was called from within `NSQD.Main()`, then it could pass `n.logf` as a parameter to that, and avoid adding the new public method. But I understand that won't be very convenient or clean, so I don't have a better idea at the moment ...